src/ode_solver/ode_solver_factory.cpp

```diff
@@ -125,6 +98,132 @@ void ODESolverFactory<dim,real,MeshType>::display_error_ode_solver_factory(Param
+                }
+                if (n_rk_stages == 4){
+                    pcout << " with 4 stages..." << std::endl;
+                    return std::make_shared<RRKExplicitODESolver<dim,real,4,MeshType>>(dg_input,rk_tableau);
+                }
+                else{
+                    pcout << "but number of stages was unclear. Aborting..." << std::endl;
+                    std::abort();
+                    return nullptr;
+                }
+            } else {
+                pcout << "RRK has only been tested on collocated nodes with Burgers. Aborting..,"<<std::endl;
+                std::abort();
+            }
+        } else {
+            pcout << "RRK has only been tested on 1D calculations. Aborting..,"<<std::endl;
+            std::abort();
+        }
+    }
+    else {
+        display_error_ode_solver_factory(ode_solver_type, false);
+        return nullptr;
+    }
+}
+
+template <int dim, typename real, typename MeshType>
+std::shared_ptr<RKTableauBase<dim,real,MeshType>> ODESolverFactory<dim,real,MeshType>::create_RKTableau(std::shared_ptr< DGBase<dim,real,MeshType> > dg_input)
+{
+    dealii::ConditionalOStream pcout(std::cout, dealii::Utilities::MPI::this_mpi_process(MPI_COMM_WORLD)==0);
+    using RKMethodEnum = Parameters::ODESolverParam::RKMethodEnum;
+    const RKMethodEnum rk_method = dg_input->all_parameters->ode_solver_param.runge_kutta_method;
+
+    const int n_rk_stages = dg_input->all_parameters->ode_solver_param.n_rk_stages;
+
+    if (rk_method == RKMethodEnum::ssprk3_ex)   return std::make_shared<SSPRK3Explicit<dim, real, MeshType>> (n_rk_stages);
+    if (rk_method == RKMethodEnum::rk4_ex)      return std::make_shared<RK4Explicit<dim, real, MeshType>>    (n_rk_stages);
+    if (rk_method == RKMethodEnum::euler_ex) {
+        using ODEEnum = Parameters::ODESolverParam::ODESolverEnum;
+        ODEEnum ode_solver_type = dg_input->all_parameters->ode_solver_param.ode_solver_type;
+        if (ode_solver_type == ODEEnum::rrk_explicit_solver) {
+            //forward Euler is invalid for RRK: sum(b_i*a_ij) = 0 (see Lemma 2.1 in Ketcheson 2019)
+            pcout << "RRK is not valid for Forward Euler. Aborting..." << std::endl;
+            std::abort();
+            return nullptr;
+        } else return std::make_shared<EulerExplicit<dim, real, MeshType>>  (n_rk_stages);
+    }
+    if constexpr(dim==1){
+        //Only tested in 1D with Burgers and advection
+        using PDEEnum = Parameters::AllParameters::PartialDifferentialEquation;
+        const PDEEnum pde_type = dg_input->all_parameters->pde_type;
+        if ((pde_type==PDEEnum::burgers_inviscid) || (pde_type==PDEEnum::advection)){
+            if (rk_method == RKMethodEnum::euler_im)    return std::make_shared<EulerImplicit<dim, real, MeshType>>  (n_rk_stages);
+            if (rk_method == RKMethodEnum::dirk_2_im)   return std::make_shared<DIRK2Implicit<dim, real, MeshType>>  (n_rk_stages);
```

Comment:
`implicit` instead of `im` in the enum members

Reply:
I chose `_ex` an `_im` for the enum members for convenience while changing parameter files. `parameters_ode_solver` has a [comment](https://github.com/dougshidong/PHiLiP/pull/169/files#diff-98dccea52bc8d9f227575d1e06bc06b192ce40bc493026b1250f9ba5110e2bddR92) explaining the suffixes. Do you think this is clear enough, or should I use the full words "explicit" and "implicit"?

Reply:
Ok that looks fine.